examples/ginkgo/ex1.cpp

```diff
@@ -194,24 +203,41 @@ int main(int argc, char *argv[])
       if (use_ginkgo_solver)
       {
 #ifdef MFEM_USE_GINKGO
-         // Solve the linear system with CG + ILU from Ginkgo.
-         std::string executor = "reference";
-         auto exec = gko::ReferenceExecutor::create();
-         auto ilu_precond =
-            gko::preconditioner::Ilu<gko::solver::LowerTrs<>,
-            gko::solver::UpperTrs<>, false>::build()
-            .on(exec);
-         GinkgoWrappers::CGSolver ginkgo_solver(executor, 1, 2000, 1e-12, 0.0,
-                                                ilu_precond.release() );
-         ginkgo_solver.solve(&((SparseMatrix&)(*A)), X, B);
+         // Solve the linear system with CG + IC from Ginkgo,
+         // or CG from Ginkgo + Gauss-Seidel from MFEM.
+         GinkgoWrappers::GinkgoExecutor exec(device);
+         if (use_ginkgo_precond) // Also use Ginkgo preconditioner.
+         {
+            GinkgoWrappers::IcPreconditioner ginkgo_precond(exec, "paric", 30);
+            GinkgoWrappers::CGSolver ginkgo_solver(exec, print_lvl, 400, 1e-12, 0.0,
+                                                   ginkgo_precond);
+            ginkgo_solver.SetOperator(*(A.Ptr()));
+            ginkgo_solver.Mult(B, X);
+         }
+         else  //Create MFEM preconditioner and wrap it for Ginkgo's use.
+         {
+            DSmoother M((SparseMatrix&)(*A));
+            GinkgoWrappers::MFEMPreconditioner gko_M(exec, M);
+            GinkgoWrappers::CGSolver ginkgo_solver(exec, print_lvl, 400, 1e-12, 0.0,
+                                                   gko_M);
+            ginkgo_solver.SetOperator(*(A.Ptr()));
+            ginkgo_solver.Mult(B, X);
+         }
 #endif
       }
+      else if (use_ginkgo_precond)  // Test using a Ginkgo preconditioner with an MFEM solver.
+      {
+         GinkgoWrappers::GinkgoExecutor exec(device);
+         GinkgoWrappers::IcPreconditioner M(exec, "paric", 30);
+         M.SetOperator(*(A.Ptr()));  // Generate the preconditioner for the matrix A.
+         PCG(*A, M, B, X, print_lvl, 400, 1e-12, 0.0);
+      }
       else
       {
 #ifndef MFEM_USE_SUITESPARSE
-         // Use a simple symmetric Gauss-Seidel preconditioner with PCG.
-         GSSmoother M((SparseMatrix&)(*A));
-         PCG(*A, M, B, X, 1, 200, 1e-12, 0.0);
+         // Use a simple Jacobi preconditioner with PCG.
+         DSmoother M((SparseMatrix&)(*A));
+         PCG(*A, M, B, X, print_lvl, 400, 1e-12, 0.0);
 #else
          // If MFEM was compiled with SuiteSparse, use UMFPACK to solve the system.
          UMFPackSolver umf_solver;
```

Comment:
This example doesn't need to have an option for a direct solver

Reply:
Agreed, removed this option.